lib/AbstractHandle/Utils/MongoUtil.py

```diff
@@ -57,14 +57,14 @@ def _get_collection(self, mongo_host, mongo_port, mongo_database, mongo_collecti
 
         return my_collection
 
-    def _inc_counter(self):
-        counter = self.get_hid_counter()
+    def increase_counter(self):
 
-        counter += 1
+        query = {'_id': self.HID_COUNTER_ID}
+        update = {'$inc': {self.HID_COUNTER_ID: 1}}
 
-        update_filter = {'_id': self.HID_COUNTER_ID}
-        update = {'$set': {self.HID_COUNTER_ID: counter}}
-        self.hid_counter_collection.update_one(update_filter, update)
+        return self.hid_counter_collection.find_one_and_update(filter=query,
+                                                               update=update,
+                                                               return_document=ReturnDocument.BEFORE)['hid_counter']
```

Comment:
shouldn't this be after? That totally prevents any weirdness causing a collision, especially the first time this goes up.

Also, shouldn't the counter upsert? I'm actually surprised this works without it.

Reply:
sure. I was trying to keep it consistent with the current version. But using AFTER and upsert doesn't hurt.